src/test/java/com/microsoft/jenkins/vmss/util/AzureUtilsTest.java

```diff
@@ -42,13 +43,13 @@ public void fromServicePrincipal() throws NoSuchFieldException, IllegalAccessExc
         );
         final ApplicationTokenCredentials token = AzureUtils.fromServicePrincipal(sp);
         final AzureEnvironment env = token.environment();
-        final Field secretField = token.getClass().getDeclaredField("secret");
-        secretField.setAccessible(true);
-        Assert.assertEquals("secret", (String)secretField.get(token));
+        final Method secretMethod = token.getClass().getDeclaredMethod("clientSecret");
```

Comment:
I don't think it is good practice to write unit tests using reflection. We should not rely on non-public API as it may change (`secret` --> `clientSecret`).

Reply:
It seems we cannot access client secret field via public API here. Do you have any suggestion for this case without sacrifice of code coverage?

Reply:
If we must test against this field, we don't have other choice, though. The check on the package visible method `clientSecret` may be a bit better.